packages/hash/shared/src/ProsemirrorManager.ts

```diff
@@ -0,0 +1,530 @@
+import { BlockVariant, JsonObject } from "@blockprotocol/core";
+import { ProsemirrorNode, Schema } from "prosemirror-model";
+import { EditorState, Transaction } from "prosemirror-state";
+import { EditorProps, EditorView } from "prosemirror-view";
+
+import {
+  fetchBlock,
+  HashBlock,
+  HashBlockMeta,
+  isTextBlock,
+  prepareBlockCache,
+} from "./blocks";
+import {
+  BlockEntity,
+  getBlockChildEntity,
+  isDraftTextContainingEntityProperties,
+  isTextEntity,
+} from "./entity";
+import {
+  createEntityStore,
+  EntityStore,
+  EntityStoreType,
+  isBlockEntity,
+  isDraftBlockEntity,
+} from "./entityStore";
+import {
+  addEntityStoreAction,
+  entityStorePluginState,
+  entityStorePluginStateFromTransaction,
+  generateDraftIdForEntity,
+  mustGetDraftEntityByEntityId,
+} from "./entityStorePlugin";
+import {
+  componentNodeGroupName,
+  findComponentNode,
+  mutateSchema,
+} from "./prosemirror";
+import { childrenForTextEntity } from "./text";
+
+type NodeViewFactory = NonNullable<EditorProps<Schema>["nodeViews"]>[string];
+
+type ComponentNodeViewFactory = (meta: HashBlockMeta) => NodeViewFactory;
+
+const isBlockCompatible = (
```

Comment:
In what circumstances would we try to load incompatible blocks?

Reply:
This isn't about loading an incompatible block. It's about whether we're being asked to render a block entity with a componentId where that component may not be compatible with the underlying data (i.e, the component ids don't match and also aren't compatible in terms of entity properties – i.e, aren't HASH text blocks). 

I'm going to look at replacing this because I can't why this would happen anymore. I think this is legacy logic from a previous iteration of block switching.

Reply:
[`f80e4ec` (#872)](https://github.com/hashintel/hash/pull/872/commits/f80e4ec6c2dbf7d766f91181067704c192ced59c)

Reply:
Makes sense, thank you